src/features/schema/fields/location/LocationForm.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select from '@mui/material/Select';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+
+import InputSlider from './InputSlider';
+import { set } from '../../../config/configSlice';
+
+export default function LocationForm({ field }) {
+    const [value, setValue] = useState({
+    	// Default to Brooklyn, because that's where tidbyt folks
+    	// are and  we can only dispatch a location object which
+    	// has all fields set.
+    	'lat': 40.6782,
+    	'lng': -73.9442,
+    	'locality': 'Brooklyn, New York',
+    	'timezone': 'America/New_York',
+    	// But overwrite with app-specific defaults set in config.
+    	...field.default
+   	});
+
+    const config = useSelector(state => state.config);
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (field.id in config) {
+            setValue(JSON.parse(config[field.id].value));
+        } else if (field.default) {
+            dispatch(set({
+                id: field.id,
+                value: field.default,
+            }));
+        }
+    }, [])
+
+    const setPart = (partName, partValue) => {
+    	let newValue = {...value};
+    	newValue[partName] = partValue;
+    	setValue(newValue);
+    	dispatch(set({
+    		id: field.id,
+    		value: JSON.stringify(newValue),
```

Comment:
👌 perfect! This will really help make sure folks understand that they need to decode this in their app